### PR TITLE
Add featuregate package and use it in controllers

### PR DESCRIPTION
**Pull request by ykulazhenkov**
This change is required to simplify usage of feature gates outside of SriovOperatorConfig controller.

 SriovOperatorConfig controller is responsible for initialization of the featuregate helper.
Other controllers can check configured feature gates by using the interface.

### CODE REVIEW

File: controllers/sriovnetworknodepolicy_controller.go
Comment (zeeke) [inline controllers/sriovnetworknodepolicy_controller.go:63]: This field is not used. I guess it will be used in some future PR, right?

File: pkg/featuregate/featuregate.go
Comment (zeeke) [inline pkg/featuregate/featuregate.go:18]: Same here, `Set` seems to be used in tests only. How is it going to be used?

File: controllers/sriovnetworknodepolicy_controller.go
Comment (ykulazhenkov) [reply controllers/sriovnetworknodepolicy_controller.go:63]: Yes, this is preparation for the next PR.

File: pkg/featuregate/featuregate.go
Comment (ykulazhenkov) [reply pkg/featuregate/featuregate.go]: @zeeke I removed `Set` function

**Comment by ykulazhenkov** [comment]
@zeeke Can we merge this PR if it looks good for you? thx

**Review by zeeke — APPROVED** [review]
LGTM

I was waiting for the CI to finish

**Pull request merged**
